ui/imports/shared/controls/chat/ProfileHeader.qml

```diff
@@ -56,6 +61,34 @@ Item {
         }
     }
 
+    Item {
+        id: tmpCroppedImageHelper
+
+        visible: false
+
+        Image {
+            id: tmpImage
+            mipmap: true
+        }
+
+        property var keepGrabResultAlive;
+
+        function setCroppedTmpIcon(source, x, y, width, height) {
+            tmpCroppedImageHelper.width = width
+            tmpCroppedImageHelper.height = height
+
+            tmpImage.x = -x
+            tmpImage.y = -y
+            tmpImage.source = source
+
+            tmpCroppedImageHelper.grabToImage(result => {
+                keepGrabResultAlive = result
```

Comment:
Are we sure `tmpCroppedImageHelper` and as a consequence `keepGrabResultAlive` will not be destroyed before we use `previewIcon`'s URL on the backend side?

Reply:
`previewIcon` is used only for preview. Backend is always provided with the full image + cropping coordinates.